pkg/route/api/validation/validation.go

```diff
@@ -15,5 +16,103 @@ func ValidateRoute(route *routeapi.Route) errs.ValidationErrorList {
+		}
+	}
+
+	//pod cert should not specify any cert
+	if tls.Termination == routeapi.TLSTerminationPassthrough {
+		if err := destTerminationCertError("certificateFile", []byte(tls.CertificateFile)); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("certificate", tls.Certificate); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("keyFile", []byte(tls.KeyFile)); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("key", tls.Key); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("caCertificateFile", []byte(tls.CACertificateFile)); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("caCertificate", tls.CACertificate); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("destinationCACertificateFile", []byte(tls.DestinationCACertificateFile)); err != nil {
+			result = append(result, err)
+		}
+
+		if err := destTerminationCertError("destinationCACertificate", tls.DestinationCACertificate); err != nil {
+			result = append(result, err)
+		}
+	}
+
+	//edge cert should specify cert, key, and cacert
+	if tls.Termination == routeapi.TLSTerminationEdge {
+		result = append(result, edgeTerminationCertError("certificate", "certificateFile", tls.Certificate, tls.CertificateFile)...)
+		result = append(result, edgeTerminationCertError("key", "keyFile", tls.Key, tls.KeyFile)...)
+		result = append(result, edgeTerminationCertError("caCertificate", "caCertificateFile", tls.CACertificate, tls.CACertificateFile)...)
+
+		if len(tls.DestinationCACertificate) > 0 {
+			result = append(result, errs.NewFieldInvalid("destinationCACertificate", "", "edge termination does not support destination certificates"))
+		}
+		if len(tls.DestinationCACertificateFile) > 0 {
+			result = append(result, errs.NewFieldInvalid("destinationCACertificateFile", "", "edge termination does not support destination certificates"))
+		}
+	}
+
+	//TODO: unsupported field
+	if len(tls.KeyPassPhrase) > 0 {
+		result = append(result, errs.NewFieldNotSupported("keyPassPhrase is not yet supported", ""))
+	}
+
+	return result
+}
+
+//helper to return a standard error for the pod term validations
+func destTerminationCertError(field string, value []byte) *errs.ValidationError {
+	if len(value) > 0 {
+		return errs.NewFieldInvalid(field, "", "specifying certificates with pod termination is not valid")
+	}
+	return nil
+}
+
+//helper to return a standard error for edge termination validations
+func edgeTerminationCertError(contentsField string, fileField string, contents []byte, file string) []error {
+	if len(contents) == 0 && len(file) == 0 {
+		err := fmt.Sprintf("edge termination requires %s or %s", contentsField, fileField)
+		return []error{errs.NewFieldInvalid(contentsField, "", err), errs.NewFieldInvalid(fileField, "", err)}
```

Comment:
Just as a note to myself, these should change to NewFieldRequired after removing the ability to specify the file name OR the full cert text.